core/src/test/java/org/apache/calcite/jdbc/CalciteRemoteDriverTest.java

```diff
@@ -880,6 +880,38 @@ public Service create(AvaticaConnection connection) {
     assertThat(updateCount, is(1));
   }
 
+  @Test public void testInsertBatchWithPreparedStatement() throws Exception {
+    final Connection connection = DriverManager.getConnection(
+        "jdbc:avatica:remote:factory="
+            + LocalServiceModifiableFactory.class.getName());
+
+    PreparedStatement pst = connection.prepareStatement(
+        "insert into \"foo\".\"bar\" values (?, ?, ?, ?, ?)");
+    pst.setInt(1, 1);
+    pst.setInt(2, 1);
+    pst.setString(3, "second");
+    pst.setInt(4, 1);
+    pst.setInt(5, 1);
+    pst.addBatch();
+    pst.addBatch();
+
+    int[] updateCounts = pst.executeBatch();
+    assertThat(updateCounts.length, is(2));
+    assertThat(updateCounts[0], is(1));
+    assertThat(updateCounts[1], is(1));
+    ResultSet resultSet = pst.getResultSet();
+    assertThat(resultSet, nullValue());
+
+    // Now empty batch
+    pst.clearBatch();
+    updateCounts = pst.executeBatch();
+    assertThat(updateCounts.length, is(0));
+    resultSet = pst.getResultSet();
+    assertThat(resultSet, nullValue());
```

Comment:
Please use assertions in such a way that failures make it clear what is verified and why.
Currently it is hard to tell which of the assertions are related to 3388, and which of them are not.

Reply:
ok, After the repair, the `PreparedStatement` can execute batch.

Reply:
Please review again.

Reply:
Please use assertions in such a way that failures make it clear what is verified and why.

Reply:
`CalciteRemoteDriverTest` using `MatcherAssert#assertThat()`.
Should we be consistent with it?